src/node_data.js

```diff
@@ -110,23 +116,24 @@ const initData = function(node, nodeName, key) {
  * @return {!NodeData} The NodeData for this Node.
  */
 const getData = function(node) {
-  let data = node['__incrementalDOMData'];
-
-  if (!data) {
-    data = importNode(node);
+  if (!node[DATA_PROP]) {
```

Comment:
Alternatively, you could just call `importNode` since it does the check for you. Probably the compiler can do it in a smart way.

We could also just call `importNode` when we enter a node, then have `getData` just return the property.

This should be fine as-is though.